backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/EligibilityController.java

```diff
@@ -44,4 +75,157 @@
 		
 	}
 
+	@PostMapping("/check-msp-coverage-status")
+	public ResponseEntity<CheckMspCoverageStatusResponse> checkMspCoverageStatus(@Valid @RequestBody CheckMspCoverageStatusRequest checkMspCoverageStatusRequest) throws HL7Exception {
+		
+		logger.info("checkMspCoverageStatus request params - phn: {}; dateOfBirth: {}; dateOfService: {}; "
+				+ "checkSubsidyInsuredService: {}; checkLastEyeExam: {}; checkPatientRestriction: {}", 
+				checkMspCoverageStatusRequest.getPhn(), checkMspCoverageStatusRequest.getDateOfBirth(), checkMspCoverageStatusRequest.getDateOfService(), checkMspCoverageStatusRequest.getCheckSubsidyInsuredService(), checkMspCoverageStatusRequest.getCheckLastEyeExam(), checkMspCoverageStatusRequest.getCheckPatientRestriction());
+				
+		E45 e45 = buildE45Message(checkMspCoverageStatusRequest.getPhn(), checkMspCoverageStatusRequest.getDateOfBirth(), checkMspCoverageStatusRequest.getDateOfService(), checkMspCoverageStatusRequest.getCheckSubsidyInsuredService(), checkMspCoverageStatusRequest.getCheckLastEyeExam(), checkMspCoverageStatusRequest.getCheckPatientRestriction());		
+		String transactionId = UUID.randomUUID().toString();		
+		Message message = eligibilityService.checkMspCoverageStatus(e45, transactionId);		
+		CheckMspCoverageStatusResponse checkMspCoverageStatusResponse = buildEligibilityResponse(message, checkMspCoverageStatusRequest.getDateOfService());		
+		ResponseEntity<CheckMspCoverageStatusResponse> response = ResponseEntity.ok(checkMspCoverageStatusResponse);
+	
+		logger.info("checkMspCoverageStatus response: {} ", checkMspCoverageStatusResponse);
+		
+		return response;
+	}
+
+    private E45 buildE45Message(String phn, LocalDate dateOfBirth, LocalDate dateOfService, Boolean checkSubsidyInsuredService, Boolean checkLastEyeExam, Boolean checkPatientRestriction) throws HL7Exception {
```

Comment:
We should likely refactor these next 6 methods into Message Converter for the next transaction right?

Reply:
As discussed, some common code available in V2MessageUtils but will add TODO to build class for converter to/from Messages.